pyccel/codegen/printing/fcode.py

```diff
@@ -2058,6 +2066,20 @@ def _print_If(self, expr):
 
         return ''.join(lines)
 
+    def _print_IfTernaryOperator(self, expr):
+        args = expr.body
+        cond = self._print(PythonBool(args[0]))
+        first = args[1]
+        second = args[2]
+        try :
+            cast_func = python_builtin_datatypes[str_dtype(expr.dtype)]
```

Comment:
Does this mean that we can't use an `IfTernaryOperator` for strings in Fortran?

Reply:
this should only cast only when types are different. for string, I did a check in the class when only one of ifternary result is a string  i raise an error otherwise if both results are strings it should work

Reply:
Ok, but this line is still called when handling Strings isn't it? So the error will still be raised. Please add a test with strings to check